File: lab02/zad04.m
n = 2e7;
u = -4.2;
sigma = 2.3;
r = u + sigma * randn(1,n);

u2 = mean(r);
sigma2 = sqrt(var(r));

bins_num = 40;
hist = histogram(r, bins_num, 'Normalization', 'pdf');

first_point = hist.BinEdges(1) + hist.BinWidth / 2;
last_point = first_point + hist.BinWidth * (bins_num - 1);

midBins = (first_point:hist.BinWidth:last_point);
pn = normpdf(midBins, u2, sigma2);

hold on
plot(midBins, pn)
hold off

File: lab02/zad04.py
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt

n = 20000000
u = -4.2
sigma = 2.3
r = u + sigma * np.random.randn(n)

u2 = np.mean(r)
sigma2 = np.sqrt(np.var(r))
print(u2)
print(sigma2)

bins_num = 50
fig, ax = plt.subplots()
num, bins, patches = ax.hist(r, bins_num, density=True)

x = np.arange(bins[0], bins[-1], 0.1)
ax.plot(x, stats.norm.pdf(x, u2, sigma2))

plt.show()
